rust/main/agents/relayer/src/settings/mod.rs

```diff
@@ -378,6 +388,43 @@ fn parse_matching_list(p: ValueParser) -> ConfigResult<MatchingList> {
     err.into_result(ml)
 }
 
+fn parse_json_object(p: ValueParser) -> Option<(ConfigPath, Value)> {
```

Comment:
we could eventually switch to this for parsing matching lists as well and remove the current complicated logic